bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/CandidateValidator.scala

```diff
@@ -1,40 +1,41 @@
 package edu.gemini.ags.impl
 
 import edu.gemini.ags.api.AgsMagnitude.MagnitudeTable
-import edu.gemini.shared.skyobject.SkyObject
-import edu.gemini.skycalc.CoordinateDiff
+import edu.gemini.spModel.core.Coordinates
+import edu.gemini.spModel.core.Target.SiderealTarget
 import edu.gemini.spModel.obs.context.ObsContext
-import edu.gemini.spModel.rich.shared.immutable._
 import edu.gemini.spModel.target.SPTarget
 import edu.gemini.ags.api.AgsMagnitude
-import edu.gemini.catalog.api.MagnitudeLimits
+import edu.gemini.catalog.api.MagnitudeConstraints
 
+import scalaz._
+import Scalaz._
 
 /**
  * Math on a list of candidates with a given set of constraints.  The idea is
  * that one set of candidates and constraints can be applied to differing
  * observation contexts (different position angles, guide speeds, etc.)
  */
-class CandidateValidator(params: SingleProbeStrategyParams, mt: MagnitudeTable, candidates: List[SkyObject]) {
+class CandidateValidator(params: SingleProbeStrategyParams, mt: MagnitudeTable, candidates: List[SiderealTarget]) {
   /**
    * Produces a predicate for testing whether a candidate is valid in an
    * established context.
    */
-  def isValid(ctx: ObsContext): (SkyObject) => Boolean = {
-    val magLimits = params.magnitudeCalc(ctx, mt).map(AgsMagnitude.autoSearchLimitsCalc(_, ctx.getConditions)).getOrElse(MagnitudeLimits.empty(params.band))
+  private def isValid(ctx: ObsContext): (SiderealTarget) => Boolean = {
+    val magLimits = params.magnitudeCalc(ctx, mt).map(AgsMagnitude.autoSearchLimitsCalc(_, ctx.getConditions)).getOrElse(MagnitudeConstraints.empty(params.band))
 
-    (so: SkyObject) => {
+    (so: SiderealTarget) => {
       // Do not use any candidates that are too close to science target / base
       // position (i.e. don't use science target as guide star)
       def farEnough =
         params.minDistance.forall { min =>
-          val soCoords = so.getHmsDegCoordinates.asSkycalc
-          val diff = new CoordinateDiff(ctx.getBaseCoordinates, soCoords)
-          diff.getDistance.compareToAngle(min) >= 0
+          val soCoords = so.coordinates
+          val diff = Coordinates.difference(ctx.getBaseCoordinates, soCoords)
+          diff.distance >= min
         }
 
       // Only keep candidates that fall within the magnitude limits.
-      def brightnessOk = so.getMagnitude(params.band).asScalaOpt.exists(magLimits.contains)
+      def brightnessOk = so.magnitudeOn(params.band).exists(m => magLimits.contains(m))
```

Comment:
Trivial: not sure if it is part of this PR but "in" seems to me a better preposition than "on" here.  As in `magnitudeIn(R)`.